tasks/build_tags.py

```diff
@@ -29,70 +32,180 @@
-IOT_AGENT_TAGS = [
-    "zlib",
+### Tag inclusion lists
+
+# AGENT_TAGS lists the tags needed when building the agent.
+AGENT_TAGS = [
+    "apm",
+    "consul",
+    "containerd",
+    "cri",
+    "docker",
+    "ec2",
+    "etcd",
+    "gce",
+    "jmx",
+    "kubeapiserver",
+    "kubelet",
+    "netcgo",
+    "process",
+    "python",
+    "secrets",
     "systemd",
+    "zk",
+    "zlib",
 ]
 
+# ANDROID_TAGS lists the tags needed when building the android agent
 ANDROID_TAGS = [
-    "zlib",
     "android",
+    "zlib",
+]
+
+# CLUSTER_AGENT_TAGS lists the tags needed when building the cluster-agent
+CLUSTER_AGENT_TAGS = [
+    "clusterchecks",
+    "kubeapiserver",
+    "orchestrator",
+    "secrets",
+    "zlib",
+]
+
+# CLUSTER_AGENT_CLOUDFOUNDRY_TAGS lists the tags needed when building the cloudfoundry cluster-agent
+CLUSTER_AGENT_CLOUDFOUNDRY_TAGS = [
+    "clusterchecks",
+    "secrets",
+]
+
+# DOGSTATSD_TAGS lists the tags needed when building dogstatsd
+DOGSTATSD_TAGS = [
+    "docker",
+    "kubelet",
+    "secrets",
+    "zlib",
+]
+
+# IOT_AGENT_TAGS lists the tags needed when building the IoT agent
+IOT_AGENT_TAGS = [
+    "systemd",
+    "zlib",
 ]
 
-PROCESS_ONLY_TAGS = [
+# PROCESS_AGENT_TAGS lists the tags necessary to build the process-agent
+PROCESS_AGENT_TAGS = AGENT_TAGS + [
+    "clusterchecks",
     "fargateprocess",
     "orchestrator",
 ]
 
+# SECURITY_AGENT_TAGS lists the tags necessary to build the security agent
+SECURITY_AGENT_TAGS = AGENT_TAGS + [
+    "clusterchecks",
+]
+
+# PROCESS_AGENT_TAGS lists the tags necessary to build system-probe
+SYSTEM_PROBE_TAGS = AGENT_TAGS + [
+    "clusterchecks",
+    "linux_bpf",
+]
+
+# TRACE_AGENT_TAGS lists the tags that have to be added when the trace-agent
+TRACE_AGENT_TAGS = [
+    "docker",
+    "kubeapiserver",
+    "kubelet",
+    "netcgo",
+    "secrets",
+]
+
+# TEST_TAGS lists the tags that have to be added to run tests
+TEST_TAGS = AGENT_TAGS + [
+    "clusterchecks",
+]
+
+### Tag exclusion lists
+
+# List of tags to always remove when not building on Linux
 LINUX_ONLY_TAGS = [
     "containerd",
     "cri",
     "netcgo",
     "systemd",
 ]
+
+# List of tags to always remove when building on Windows
+WINDOWS_EXCLUDE_TAGS = [
+    "linux_bpf"
+]
+
+# List of tags to always remove when building on Windows 32-bits
 WINDOWS_32BIT_EXCLUDE_TAGS = [
-    "orchestrator",
     "docker",
     "kubeapiserver",
     "kubelet",
+    "orchestrator",
 ]
 
-
-def get_default_build_tags(iot=False, process=False, arch="x64", android=False):
+def get_default_build_tags(build="agent", arch="x64"):
     """
-    Build the default list of tags based on the current platform.
+    Build the default list of tags based on the build type and current platform.
 
     The container integrations are currently only supported on Linux, disabling on
     the Windows and Darwin builds.
     """
-    include = ["all"]
-    if iot:
+    include = []
```

Comment:
These ifs could be `elif` and the default `include = []` in a final `else` (and maybe print a warning, since we don't expect the default to be ever used?).